src/Apps/Legion/Components/WizardShell/Internal/ModifyStep/Internal/TypesTab/Internal/TypesCard/TypesCard.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useMemo } from 'react';
+import { getDebugLogger } from '../../../../../../../../../../Models/Services/Utils';
+import { ITypesCardProps } from './TypesCard.types';
+import { getStyles } from './TypesCard.styles';
+import { useTypes } from '../../../../../../../../Hooks/useTypes';
+import {
+    DetailsList,
+    IColumn,
+    TextField,
+    SelectionMode,
+    Checkbox
+} from '@fluentui/react';
+import { useEntities } from '../../../../../../../../Hooks/useEntities';
+import TypeIcon from '../../../../../../../TypeIcon/TypeIcon';
+import { IViewProperty } from '../../../../../../../../Models';
+import { useExtendedTheme } from '../../../../../../../../../../Models/Hooks/useExtendedTheme';
+import { useTranslation } from 'react-i18next';
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('TypesCard', debugLogging);
+
+const getColumns = (): IColumn[] => [
+    {
+        key: 'check-box-column',
+        name: '',
+        minWidth: 32,
+        maxWidth: 32,
+        onRender: (item: IViewProperty) => {
+            return (
+                <Checkbox
+                    onChange={(_ev, checked) => {
+                        // TODO: Integrate this with Mitch's changes
+                        // checked ? deleteProperty(item.id) : addProperty(item);
+                        // eslint-disable-next-line no-console
+                        console.log(checked);
+                    }}
+                    defaultChecked={!item.isDeleted}
+                />
+            );
+        }
+    },
+    {
+        key: 'source-name-column',
+        name: 'Source property name',
+        minWidth: 100,
+        fieldName: 'sourcePropId'
+    },
+    {
+        key: 'friendly-name-column',
+        name: 'Friendly name',
+        minWidth: 200,
+        fieldName: 'friendlyName'
+    }
+];
+
+const TypesCard: React.FC<ITypesCardProps> = (props) => {
```

Comment:
I see that TypesCard is an internal element for TypesTab. But it creates two level of /Internal folder structure, can we make single level /Internal meaning moving the TypesCard one level above as a sibling to TypesTab internal component. The reason is that it is only one scingle component under second /Internal which I think does not require deep nesting.

Reply:
I think infinite nesting of internal components allows us to keep the same pattern regardless of depth. I like the consistency, but we can discuss if it is something that you would rather not have.